client/src/pages/VocabDrawer/VocabDrawer.js

```diff
@@ -476,7 +522,11 @@ const VocabDrawer = ({ navigation }) => {
         stopRecording={stopRecording}
         playRecording={playRecording}
         confirmRecording={confirmRecording}
-        discardRecording={clearRecording}
+        discardRecording={() => {
```

Comment:
I would create a helper function above with all of this logic, instead of defining the function within the prop list.